Applications/Spire/Source/Ui/RegionBox.cpp

```diff
@@ -0,0 +1,202 @@
+    RegionBox::get_query_model() const {
+  return m_query_model->m_source;
+}
+
+const std::shared_ptr<RegionBox::CurrentModel>& RegionBox::get_current() const {
+  return m_current;
+}
+
+void RegionBox::set_placeholder(const QString& placeholder) {
+  m_tag_combo_box->set_placeholder(placeholder);
+}
+
+bool RegionBox::is_read_only() const {
+  return m_tag_combo_box->is_read_only();
+}
+
+void RegionBox::set_read_only(bool is_read_only) {
+  m_tag_combo_box->set_read_only(is_read_only);
+}
+
+connection RegionBox::connect_submit_signal(
+    const SubmitSignal::slot_type& slot) const {
+  return m_submit_signal.connect(slot);
+}
+
+bool RegionBox::eventFilter(QObject* watched, QEvent* event) {
+  if(watched == m_tag_combo_box && event->type() == QEvent::LayoutRequest) {
+    resize(m_tag_combo_box->size());
+  } else if(watched == m_tag_combo_box && event->type() == QEvent::Move) {
+    if(!m_is_external_move) {
+      m_is_tag_combo_box_move = true;
+      move(parentWidget()->mapFromGlobal(mapToGlobal(m_tag_combo_box->pos())));
+      m_is_tag_combo_box_move = false;
+    }
+  } else if(watched == parentWidget() && event->type() == QEvent::Move) {
+    QApplication::sendEvent(this, event);
+  }
+  return QWidget::eventFilter(watched, event);
+}
+
+bool RegionBox::event(QEvent* event) {
+  if(event->type() == QEvent::LayoutRequest) {
+    if(m_tag_combo_box->minimumSize() != minimumSize()) {
+      m_tag_combo_box->setMinimumSize(minimumSize());
+    }
+    if(m_tag_combo_box->maximumSize() != maximumSize()) {
+      m_tag_combo_box->setMaximumSize(maximumSize());
+    }
+  }
+  return QWidget::event(event);
+}
+
+void RegionBox::showEvent(QShowEvent* event) {
+  if(auto parent = parentWidget()) {
+    if(auto layout = parent->layout()) {
+      if(layout->indexOf(this) >= 0) {
+        parent->installEventFilter(this);
+      }
+    }
+  }
+  QWidget::showEvent(event);
+}
+
+void RegionBox::moveEvent(QMoveEvent* event) {
+  if(!m_is_tag_combo_box_move) {
+    m_is_external_move = true;
+    QApplication::sendEvent(m_tag_combo_box, event);
+    m_is_external_move = false;
+  }
+  QWidget::moveEvent(event);
+}
+
+void RegionBox::resizeEvent(QResizeEvent* event) {
+  if(m_tag_combo_box->minimumSize() != minimumSize()) {
+    m_tag_combo_box->setMinimumSize(minimumSize());
+  }
+  if(m_tag_combo_box->maximumSize() != maximumSize()) {
+    m_tag_combo_box->setMaximumSize(maximumSize());
+  }
+  QWidget::resizeEvent(event);
+}
+
+void RegionBox::on_current(const Region& region) {
+  while(m_tag_combo_box->get_current()->get_size() != 0) {
+    m_tag_combo_box->get_current()->remove(
+      m_tag_combo_box->get_current()->get_size() - 1);
+  }
+  m_tag_combo_box->get_current()->push(region);
+}
+
+void RegionBox::on_submit(const std::shared_ptr<AnyListModel>& submission) {
+  auto region = Nexus::Region();
+  for(auto i = 0; i < submission->get_size(); ++i) {
+    region = region + std::any_cast<Region&&>(submission->get(i));
```

Comment:
Does ```+=``` work?

Reply:
No, 'Nexus::Region' does not define `+=` operation.